typeDefs.js

```diff
@@ -0,0 +1,187 @@
+const { gql } = require('apollo-server');
+const { newsType, newsInputType } = require('./modules/news/news.model');
+const {
+  categoryType,
+  categoryInput,
+} = require('./modules/categories/categories.model');
+const {
+  materialType,
+  materialInput,
+} = require('./modules/materials/materials.model');
+const {
+  patternsType,
+  patternsInput,
+} = require('./modules/patterns/patterns.model');
+const {
+  currencyType,
+  currencyInput,
+} = require('./modules/currencies/currencies.model');
+
+const typeDefs = gql`
+  type Language {
+    lang: String!
+    value: String!
+  }
+
+  type ImageSet {
+    large: String
+    medium: String
+    small: String
+    thumbnail: String
+  }
+
+  type PrimaryImage {
+    primary: ImageSet!
+    additional: [ImageSet!]
+  }
+  input PrimaryImageInput {
+    primary: ImageSetInput!
+    additional: [ImageSetInput!]
+  }
+
+  type ConvertOption {
+    name: String!
+    exchangeRate: Float!
+  }
+ 
+  type Currency {
+    ${currencyType}
+  }
+
+  type Subcategory {
+    _id: ID!
+    categoryCode: String!
+    name: [Language!]
+    images: ImageSet
+    available: Boolean!
+  }
+
+  type Category {
+    ${categoryType}
+  }
+
+  type Author {
+    name: String!
+    image: ImageSet
+  }
+
+  input AuthorInput {
+    name: String!
+    image: ImageSetInput
+  }
+
+  type News {
+  ${newsType}
+  }
+input NewsInput {
+${newsInputType}
+}
+
+  type Color {
+    code: Int!,
+    name: [Language!],
+    images: ImageSet,
+    available: Boolean!,
+  }
+
+  type Material {
+    ${materialType}
+  }
+
+  type Pattern {
```

Comment:
Can we move not only body to the separate file, but also type `Pattern` too

Reply:
yes we can. Done.